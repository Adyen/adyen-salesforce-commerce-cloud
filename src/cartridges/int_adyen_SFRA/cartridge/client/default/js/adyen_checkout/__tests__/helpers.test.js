/**
 * @jest-environment jsdom
 */
const { paymentFromComponent } = require('../helpers');

let component;
beforeEach(() => {
  component = {
    handleAction: jest.fn(),
    setStatus: jest.fn(),
    reject: jest.fn(),
  };
});

describe('Helpers', () => {
  it('should make payment ajax call with fullResponse', async () => {
    document.body.innerHTML = `
        <div id="adyenPaymentMethodName"></div>
        <form id="showConfirmationForm"></form>
      `;
    const data = {
      fullResponse: { action: 'mocked_action' },
      paymentMethod: 'mocked_paymentMethod',
    };
    $.ajax = jest.fn(({ success }) => {
      success(data);
      return { fail: jest.fn() };
    });
    await paymentFromComponent(data, component);
    expect(component.handleAction).toBeCalledWith(data.fullResponse.action);
  });

  it('  should make payment ajax call that fails', async () => {
    document.body.innerHTML = `
        <div id="adyenPaymentMethodName"></div>
        <form id="showConfirmationForm"></form>
      `;
    window.HTMLFormElement.prototype.submit = jest.fn();
    const data = { data: {}, paymentMethod: 'mocked_paymentMethod' };
    $.ajax = jest.fn(({ success }) => {
      success({});
      return { fail: jest.fn() };
    });
    await paymentFromComponent(data, component);
    expect(data).toMatchSnapshot();
  });
});
