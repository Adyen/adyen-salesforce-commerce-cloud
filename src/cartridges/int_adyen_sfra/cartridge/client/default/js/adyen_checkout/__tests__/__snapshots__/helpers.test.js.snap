// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Helpers   should make payment ajax call that fails 1`] = `
Object {
  "data": Object {},
  "paymentMethod": "mocked_paymentMethod",
}
`;
